pkg/finder/finder.go

```diff
@@ -294,19 +323,44 @@ func (f *criFinder) updateInfo() error {
 	return nil
 }
 
-func getDevices(envs []*runtime.KeyValue) map[devicePluginResourceName][]*DeviceInfo {
+func (f *finder) getDevices(podName, containerName string, envs []*runtime.KeyValue) (map[devicePluginResourceName][]*DeviceInfo, error) {
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	//Pod Resource API client
+	resp, err := f.podResourceClient.List(ctx, &podresourcesapi.ListPodResourcesRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("%v.Get(_) = _, %v", f.podResourceClient, err)
+	}
 	devices := make(map[devicePluginResourceName][]*DeviceInfo)
-	for _, env := range envs {
-		if !strings.HasPrefix(env.Key, "example.com/device") {
+	for _, podResource := range resp.GetPodResources() {
+		if podResource.Name != podName {
 			continue
 		}
-		k := strings.Split(env.Key, "_")
-		devInfo := NewDeviceInfo(k[1], k[2], env.Value)
-		var devPluginName devicePluginResourceName
-		devPluginName = devicePluginResourceName(k[0])
-		devices[devPluginName] = append(devices[devPluginName], devInfo)
+		for _, container := range podResource.GetContainers() {
+			if container.Name != containerName {
+				continue
+			}
+			for _, device := range container.GetDevices() {
+				for _, devId := range device.GetDeviceIds() {
+					for _, env := range envs {
+						if !strings.HasPrefix(env.Key, "example.com/device") {
```

Comment:
Deliberately have "example.com/device" to match the master branch of sample device plugin, will update this when the sample device plugin PR updating the environment variable gets merged into master